v2-migration/src/main/java/software/amazon/awssdk/v2migration/TransferManagerMethodsToV2.java

```diff
@@ -95,10 +111,100 @@ public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, Execu
+            }
 
             return super.visitMethodInvocation(method, executionContext);
         }
 
+        @Override
+        public J visitNewClass(J.NewClass newClass, ExecutionContext executionContext) {
+            JavaType type = newClass.getType();
+            if (!(type instanceof JavaType.FullyQualified)) {
+                return newClass;
+            }
+            if (type.isAssignableFrom(S3_TM_CREDENTIAL) &&
+                newClass.getArguments().size() == 1 &&
+                newClass.getArguments().get(0).getType() != null) {
+                if (newClass.getArguments().get(0).getType().isAssignableFrom(V2_AWSCREDENTAIL)) {
+                    addS3AsyncClientImport();
+                    addStaticCredentialsProviderImport();
+
+                    return JavaTemplate
+                        .builder("S3TransferManager.builder()" +
+                                 ".s3Client(S3AsyncClient.builder()" +
+                                 ".credentialsProvider(StaticCredentialsProvider.create(#{any()}))" +
+                                 ".build())" +
+                                 ".build()")
+                        .build()
+                        .apply(getCursor(), newClass.getCoordinates().replace(), newClass.getArguments().get(0));
+                }
+                if (newClass.getArguments().get(0).getType().isAssignableFrom(V2_CREDENTIAL_PROVIDER)) {
+                    addS3AsyncClientImport();
+
+                    return JavaTemplate
+                        .builder("S3TransferManager.builder()" +
+                                 ".s3Client(S3AsyncClient.builder()" +
+                                 ".credentialsProvider(#{any()})" +
+                                 ".build())" +
+                                 ".build()")
+                        .build()
+                        .apply(getCursor(), newClass.getCoordinates().replace(), newClass.getArguments().get(0));
+
+                }
+            }
+
+            return super.visitNewClass(newClass, executionContext);
+        }
+
+        private J.MethodInvocation transformResumeDownload(J.MethodInvocation method) {
+            String v2Method = "#{any()}.resumeDownloadFile(#{any()})";
+
+            method = JavaTemplate.builder(v2Method).build()
+                                 .apply(getCursor(), method.getCoordinates().replace(), method.getSelect(),
+                                        method.getArguments().get(0));
+            return method;
+        }
+
+        private J.MethodInvocation transformResumeUpload(J.MethodInvocation method) {
+            String v2Method = "#{any()}.resumeUploadFile(#{any()})";
+
+            method = JavaTemplate.builder(v2Method).build()
+                                 .apply(getCursor(), method.getCoordinates().replace(), method.getSelect(),
+                                        method.getArguments().get(0));
+            return method;
+        }
```

Comment:
For these name change only transforms, should we just use ChangeMethodName yaml recipe like https://github.com/aws/aws-sdk-java-v2/blob/master/v2-migration/src/main/resources/META-INF/rewrite/change-s3-methods.yml ?